feat: added mines game backend, modified backend folder structure and…

nimit9 opened the pull request:
Added Stakes Mines Game (https://stake.com/casino/games/mines)

- [x] Backend
- [ ] Frontend 

Also added provably fairness to the games, to generate randomness using [`crypto`](https://www.npmjs.com/package/crypto)

hkirat committed:
Merge pull request #3 from SauravBeginner/ui_fixed

UI fixed route added mobile responsive

nimit9 committed:
feat: added mines game backend, modified backend folder structure and apis

nimit9 committed:
feat: added mines game backend, modified backend folder structure and apis

nimit9 commented:
@hkirat Need to add auth, db, zod
Frontend: tailwind, shadcn?
Maybe monorepofy it as well?

nimit9 committed:
feat: added limbo game

nimit9 committed:
feat: added limbo game

nimit9 commented:
Added Limbo backend

nimit9 committed:
fix: dont need to pass cursor

nimit9 committed:
fix: dont need to pass cursor

nimit9 committed:
fix: modified Plinkoo to use rng seeding

nimit9 committed:
fix: modified Plinkoo to use rng seeding

nimit9 commented:
Modified Plinkoo RNG

nimit9 committed:
added dice and keno backend

nimit9 committed:
added dice and keno backend

nimit9 commented:
Added Dice and Keno

nimit9 commented:
![image](https://github.com/user-attachments/assets/43fda235-3f40-4193-b048-baca5d9dfb72)

hkirat commented:
Can yu add a video of these games?

hkirat commented:
oh backend only. hmm
Do you want to try the frontend in a different PR?

nimit9 commented:
yes, will add it in this PR itself

Devilla commented:
@here I would like to work on the frontend and modify backend with the verification algorithms a bit. [Reference](https://github.com/Devilla/provably-fair/blob/master/src/services/provably-fair/generic.ts)

Devilla commented:
Unable to get `http://localhost:3000/api/v1/rng/get-server-seeds`

Screenshot:
<img width="1055" alt="Screenshot 2024-09-09 at 12 22 26 AM" src="https://github.com/user-attachments/assets/de487d32-5059-492b-ba1e-124b40351299">

nimit9 commented:
> @here I would like to work on the frontend and modify backend with the verification algorithms a bit. [Reference](https://github.com/Devilla/provably-fair/blob/master/src/services/provably-fair/generic.ts)

Already workingg on it, but you are free to do it as well

Devilla commented:
> > @here I would like to work on the frontend and modify backend with the verification algorithms a bit. [Reference](https://github.com/Devilla/provably-fair/blob/master/src/services/provably-fair/generic.ts)
> 
> Already workingg on it, but you are free to do it as well

Sounds good, then I'll start working on dice frontend, can I raise PR for that on your forked repo so the changes are included here.

nimit9 commented:
I'm done with dice FE

Devilla commented:
> I'm done with dice FE

Can review the PR, and push Dice FE.

Devilla commented:
Looks good? Needs suggestions if any
<img width="1462" alt="Screenshot 2024-09-09 at 4 46 46 PM" src="https://github.com/user-attachments/assets/c2c93fcf-0653-4f3d-aaf0-2d3f11918e8b">

nimit9 commented:
[Screencast from 2024-09-08 20-27-27.webm](https://github.com/user-attachments/assets/7d1ec74b-e231-4e8b-bb56-ff38289b06ad)

Devilla commented:
> Screencast.from.2024-09-08.20-27-27.webm

Great design implementation, would you be open to some good freelance gigs, Lmk. 😊

nimit9 committed:
feat: initialize project structure with frontend and backend setup, including game utilities and configurations

nimit9 committed:
chore: remove obsolete audio assets including images, videos, and documents

nimit9 committed:
Merge branch 'main' of github.com:nimit9/plinkoo

nimit9 committed:
Revert "Merge branch 'main' of github.com:nimit9/plinkoo"

This reverts commit 2c96fd125dde1b60f571af739f736de7cd610c1e, reversing
changes made to 523dbcf21ff032c52b751a53da6056bec5b3ebb6.

nimit9 committed:
chore: update pnpm-lock.yaml with new Radix UI packages and versions; enhance VSCode settings for ESLint and formatting; modify TypeScript configuration for module resolution; refactor server setup to include user routes; improve error handling in passport configuration; add user management features including balance retrieval and seed rotation; update database schema for betting functionality; clean up unused routes and components in frontend.

nimit9 committed:
feat: enhance dice betting functionality by adding bet amount validation, user balance checks, and payout calculation in placeBet method

nimit9 committed:
refactor: standardize import statements and formatting in routeTree.gen.ts for improved readability

nimit9 committed:
feat: implement provably fair system with seed management, verification inputs, and calculation breakdown for enhanced user experience in dice game

nimit9 committed:
feat: add functionality to reveal server seed by its hash, including new API endpoint, frontend components, and database schema updates for hashed server seed management

nimit9 committed:
refactor: remove RevealedSeed component and update database schema with new index for hashed server seed management

nimit9 committed:
fix: update button styling in UnhashServerSeed component for improved user interface

nimit9 committed:
feat: add fetch bets list api and added ui on FE

nimit9 committed:
refactor: update betting functionality to handle amounts in cents, ensuring consistency in balance checks and payout calculations across the application

nimit9 committed:
feat: implement roulette game components including betting controls, chip management, and roulette wheel with drag-and-drop functionality; enhance UI with new styles and SVG assets

nimit9 committed:
refactor: remove unnecessary React imports from multiple components to streamline code and improve performance

nimit9 committed:
feat: implement new betting components and enhance drag-and-drop functionality in roulette game, including BottomBets, BottomColorBets, BottomNumberBets, and DroppableArea for improved user interaction

nimit9 committed:
refactor: simplify betting component labels and enhance hover effects in roulette game for improved user experience

nimit9 committed:
feat: enhance roulette betting functionality with improved state management, validation, and user interaction; refactor components for better performance and maintainability

nimit9 committed:
fix: correct betId assignment in BottomNumberBets component and remove unused selection validation in roulette game to enhance functionality and streamline code

nimit9 committed:
refactor: streamline imports and enhance type usage across various components for improved code clarity and maintainability

nimit9 committed:
refactor: remove balance store and transition to react-query for balance management across components, enhancing state consistency and reducing complexity

nimit9 committed:
feat: integrate framer-motion for enhanced animations in roulette game components, improve betting functionality with multiplier support, and refine user interactions across various betting elements

nimit9 committed:
feat: implement mines game functionality with enhanced validation, user balance checks, and improved game state management; refactor service methods for better integration with database and user management

nimit9 committed:
feat: add mines game integration with enhanced API responses, user authentication, and improved game state management; refactor service methods for better database interaction and validation

nimit9 committed:
Setup husky and lint-staged for pre-commit checks

nimit9 committed:
refactor: update user balance handling to store as string and improve balance retrieval methods across games; enhance validation checks for betting functionality

nimit9 committed:
refactor: standardize code formatting and improve consistency across various components; update package.json scripts for database generation and enhance development setup

nimit9 committed:
chore: add GitHub Actions workflow for linting and formatting on pull requests to ensure code quality and consistency

nimit9 committed:
feat: add GitHub Actions workflow for deploying frontend to S3 and CloudFront, including build steps, environment variable setup, and cache management for optimized performance

nimit9 committed:
chore: remove cache configuration for PNPM in frontend deployment workflow to streamline setup

nimit9 committed:
feat: add GitHub Actions workflow for deploying backend to production, including Docker image build, EC2 deployment, and notification upon success

nimit9 committed:
feat: add Dockerfile for API service setup, including Node.js environment, package installation, and Prisma client generation

nimit9 committed:
feat: add TypeScript configuration file for compiler options and module resolution settings

nimit9 committed:
fix: update SSH configuration in backend deployment workflow to use standardized secret names for improved clarity and consistency

nimit9 committed:
feat: add build step for API service in Dockerfile to ensure proper compilation before starting the application

nimit9 committed:
chore: add .dockerignore file to exclude unnecessary files from Docker context and update package name from 'testimonials' to 'simcasino'; adjust Node.js version in deployment workflow to latest LTS

nimit9 committed:
chore: update Dockerfile.api to install OpenSSL 1.1 and other required packages for Prisma compatibility

nimit9 committed:
chore: update Dockerfile.api to use node:22-slim and install OpenSSL with apt-get for improved compatibility and reduced image size

nimit9 committed:
chore: add VITE_APP_API_URL to environment variables in turbo.json for improved configuration management

nimit9 committed:
chore: add vercel.json configuration file for routing setup in frontend deployment

nimit9 committed:
chore: update deployment workflows to trigger on 'dont-deploy' branch instead of 'main' for both backend and frontend

nimit9 committed:
chore: update text sizes in various components for improved readability and consistency

nimit9 committed:
chore: add db:migrate script to package.json and update packageManager version to pnpm@9.9.0; create .env.example for database configuration

nimit9 committed:
feat(keno): implement keno game functionality and UI components

- Added Keno game to the frontend, including game routing and components.
- Created BettingControls and KenoTile components for user interaction.
- Integrated keno risk selection and payout multipliers.
- Implemented tile selection logic with auto-pick feature.
- Established state management for keno using Zustand.
- Defined constants and types for keno payouts and risks.
- Added SVG assets for keno game visuals.
- Updated Tailwind CSS configuration for keno-specific styles.

nimit9 committed:
feat(keno): implement betting functionality with user balance checks and transaction handling

nimit9 committed:
feat(keno): enhance betting functionality with request validation and outcome management; update UI components for better user experience

nimit9 committed:
feat(keno): add Keno game support with UI components and verification logic; enhance fairness modal and result breakdown

nimit9 committed:
feat(blackjack): implement initial Blackjack game functionality with betting and game state management; add routes and controllers for game actions

nimit9 committed:
feat(blackjack): add Blackjack game implementation with betting controls, game state management, and SVG assets

nimit9 committed:
feat(blackjack): implement comprehensive Blackjack game features including betting, game state management, and player actions; add validation schemas and update database schema

nimit9 committed:
feat: add playing card component and assets for blackjack game

- Introduced PlayingCard component for rendering cards with front and back faces.
- Added card images (C.png, D.png, H.png, S.png, back.png) to the assets.
- Updated BettingControls to handle insurance actions and improved game state management.
- Refactored getValidActionsFromState to accept active state for better action validation.
- Enhanced blackjack store to manage flipped and incoming cards.
- Cleaned up unused code in Blackjack component and improved styling.

nimit9 committed:
feat(user): add active bet check in rotateSeed to prevent seed rotation during active bets

nimit9 committed:
feat: update page title to 'Sim Casino' in index.html

nimit9 committed:
feat: add react-hot-toast for notification handling and implement error notifications in Seeds component

nimit9 committed:
feat(blackjack): implement game table and betting controls; enhance card animations and state management

nimit9 committed:
feat(blackjack): add unique IDs to cards in game state and actions for better tracking

nimit9 committed:
feat(blackjack): enhance card and game state management with new card borders and game result calculations

nimit9 committed:
feat: Add landing page with hero section, featured games, core features, transparency feed, and footer

feat: Implement transparency feed component to display real-time game results with statistics

refactor: Extract MyBets table logic into a separate component for better organization

style: Update CSS variables for improved theming and add new gradient and shadow styles

fix: Update route tree to include public index route for landing page

feat: Create home route for protected casino section

feat: Add public index route to render landing component

style: Enhance Tailwind configuration with new animations and transition properties

feat: Update blackjack types and utils to include payout information

feat: Extend user types to include optional user details in paginated bet data

nimit9 committed:
feat(betting): implement centralized betting middleware and utilities for validation and transaction handling

nimit9 committed:
feat(header): wrap logo image in Link component for improved navigation

nimit9 committed:
feat(mines): enhance game logic with user instance handling and validation middleware for play round requests

nimit9 committed:
feat(keno): refactor placeBet to use centralized transaction handling and add bet validation middleware

nimit9 committed:
feat(roulette): implement validation middleware for roulette bets and refactor placeBetAndSpin logic

nimit9 committed:
feat: add responsive viewport handling and update betting tables with viewport-aware columns

nimit9 committed:
feat(mines): update tile sizes for responsiveness and enhance layout structure in Mines component

nimit9 committed:
feat(mines): update tile sizes for responsiveness and enhance layout structure in Mines component

nimit9 committed:
feat(keno): enhance responsive design for BettingControls and KenoTile components

nimit9 committed:
feat(roulette): enhance component structure and responsiveness in RouletteTable and related components

nimit9 committed:
feat: enhance responsiveness and layout adjustments in various components

nimit9 committed:
feat(blackjack): add result breakdown and preview components; enhance game verification

- Implemented  and  components for displaying game results.
- Updated  to include blackjack results.
- Enhanced  to handle blackjack outcomes.
- Refactored game result queries to include specific keys for blackjack, keno, and dice.
- Improved card positioning logic in  and  components for better responsiveness.
- Added utility functions for calculating card positions based on viewport type.
- Cleaned up console logs and adjusted styles for better UI consistency.

nimit9 committed:
feat: enhance provably fair documentation and routing

- Added new documentation components for better structure and readability.
- Implemented Conversions, GameEvents, Implementation, and Overview pages for provably fair features.
- Updated routing to include new pages under the provably fair section.
- Improved UI elements in DiceResultPreview and ProvablyFairCalculation components.
- Introduced dropdown menu for navigation in the ProvablyFair component.
- Removed unnecessary console logs and adjusted layout styles for better UX.
- Updated Tailwind CSS configuration for improved styling.

nimit9 committed:
feat: add Bets, Logout, and User icons; enhance Header with dropdown menu for My Bets and Logout functionality

nimit9 committed:
feat: update turbo.json formatting and add BetData interface

- Reformatted the  file for consistency.
- Introduced a new  interface in  to encapsulate bet-related data structure.
- Updated  to include new dependencies and versions for Radix UI components and Prettier plugins.

nimit9 committed:
feat(dice): implement DiceBetViz component and integrate game controls; enhance game visualization in BetModal

nimit9 committed:
feat(mines): implement MinesBetViz and MinesContainer components; enhance game visualization in GameViz

nimit9 committed:
feat(bets): filter inactive bets in getTopBets and getUserBets functions

nimit9 committed:
feat(auth): remove deprecated authentication middleware

nimit9 committed:
feat(games): throw NotFoundError for inactive games in blackjack and mines controllers
feat(auth): throw UnAuthenticatedError for unauthorized access in auth middleware

nimit9 committed:
refactor(auth): replace requireAuth middleware with isAuthenticated in dice and mines routers

nimit9 committed:
feat(blackjack, keno): add BlackjackBetViz and KenoBetViz components; refactor GameViz to utilize new components

nimit9 committed:
feat: enhance betting functionality and user experience

- Updated getBet controller to include user-specific bet verification.
- Added middleware to verify user authentication for bet retrieval.
- Modified getBetById service to accept user context for personalized data.
- Enhanced user service to include active bets in user instance.
- Implemented new overview component for provable fairness documentation.
- Improved fairness modal to handle game-specific logic and navigation.
- Updated global modals to manage fairness and bet modals more effectively.
- Refactored verification inputs to support numeric nonce handling.
- Added support for active games and rotation logic in fairness components.
- Updated frontend components to improve styling and usability.
- Introduced zod schemas for better type safety in routing.
- Updated package dependencies to include the latest Prisma client.

nimit9 committed:
feat: update bet handling and remove roulette references across components

nimit9 committed:
feat: implement rate limiting for betting endpoints and add Redis support

nimit9 committed:
refactor: rename server variable to app and improve graceful shutdown handling

nimit9 committed:
refactor: remove Redis store configuration from rate limiting middleware for development use

nimit9 committed:
refactor: update deployment workflow for AWS EC2 with improved naming and comments

nimit9 committed:
refactor: enhance Dockerfile for improved build process and security practices

nimit9 committed:
refactor: update Dockerfile to copy Prisma schema from the new database package location

nimit9 committed:
refactor: remove console.log statements from game visualization components

nimit9 committed:
refactor: comment out push trigger and paths in backend deployment workflow

nimit9 committed:
refactor: comment out local authentication routes in auth.router.ts

nimit9 committed:
refactor: update keyGenerator to use ipKeyGenerator for improved request handling

nimit9 committed:
Refactor authentication flow and remove unused components

- Updated LoginModal to use LOGIN_URL constant for Google OAuth redirection.
- Created LOGIN_URL in immutableConst for better maintainability.
- Removed deprecated Login component from auth feature.
- Adjusted Home component to conditionally render tabs based on user authentication state.
- Enhanced MyBetsTable to display an empty state with action to explore games.
- Integrated Header component into Protected and Public layouts for consistent navigation.
- Removed unused login route and related code from routing structure.
- Added EmptyState component for better handling of empty data scenarios.

nimit9 committed:
feat: add backend deployment workflow for DigitalOcean

nimit9 committed:
refactor: update backend deployment workflow for DigitalOcean

nimit9 committed:
refactor: update backend deployment workflow to AWS EC2 and improve deployment script

nimit9 committed:
fix: change exposed port from 3000 to 5000 in Dockerfile.api

nimit9 committed:
feat: enhance DigitalOcean deployment with environment variables and restart policy

nimit9 committed:
fix: update CMD in Dockerfile.api to point to the correct entry file

nimit9 committed:
refactor: streamline Dockerfile.api by removing dev dependencies and enhancing layer caching

nimit9 committed:
refactor: simplify Dockerfile.api by removing multi-stage build and unnecessary dependencies

nimit9 committed:
feat: enhance Google OAuth integration with improved error handling and dynamic callback URL

nimit9 committed:
refactor: improve code readability in server.ts by formatting comments and indentation

nimit9 committed:
refactor: format fetchPost call in cashOut function for improved readability

nimit9 committed:
feat: add Privacy Policy and Terms and Conditions pages with structured content

nimit9 committed:
feat: add Privacy Policy and Terms and Conditions routes

nimit9 committed:
feat: Update Privacy Policy and Terms and Conditions pages

- Refactored Privacy Policy and Terms and Conditions components to include Header and container styling.
- Added new routes for Privacy Policy and Terms and Conditions in the route tree.
- Adjusted route definitions to remove trailing slashes for consistency.

nimit9 committed:
feat: update featured games list with Mines and Blackjack details

nimit9 committed:
feat: add showCloseIcon prop to DialogContent for conditional close button rendering
